BankAccount/pkg/tests/transactionhistorystore_test.go

```diff
@@ -0,0 +1,147 @@
+package tests
+
+import (
+	"bankacc/pkg/entities"
+	"bankacc/pkg/store"
+	"database/sql"
+	"log"
+	"testing"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/stretchr/testify/suite"
+)
+
+func MySQLInit() *sql.DB {
+	dbDriver := "mysql"
+	dbUser := "root"
+	dbPass := "Password1!"
+	dbName := "BankAccount?parseTime=true"
+	db, err := sql.Open(dbDriver, dbUser+":"+dbPass+"@/"+dbName)
+	if err != nil {
+		log.Println(err)
+	}
+	return db
+}
+
+type TearDownTestSuite interface {
+	TearDownTest()
+}
+
+type TransactionHistoryTestSuite struct {
+	suite.Suite
+	Transaction      *entities.TransactionHistory
+	Transactions     []entities.TransactionHistory
+	TransactionsP    []*entities.TransactionHistory
+	TransactionStore store.TransactionHistoryModel
+	Db               *sql.DB
+}
+
+func (suite *TransactionHistoryTestSuite) SetupTest() {
+	var err error
+	suite.Db = MySQLInit()
+	transaction := store.NewTransactionHistoryStoreModel(suite.Db)
+	suite.Transactions = []entities.TransactionHistory{
+		{
+			UserId:    1,
+			AccountId: 1,
+			Amount:    100,
+			Action:    "Deposit",
+		},
+		{
+			UserId:    2,
+			AccountId: 1,
+			Amount:    200,
+			Action:    "Withdraw",
+		},
+		{
+			UserId:    3,
+			AccountId: 3,
+			Amount:    300,
+			Action:    "Deposit",
+		},
+	}
+
+	for _, value := range suite.Transactions {
+		suite.Transaction, err = transaction.Insert(value.UserId, value.AccountId, value.Amount, value.Action)
+		if err != nil {
+			suite.T().Fatal("Unable to run InsertTransactionHistory store func")
+		}
+		suite.TransactionsP = append(suite.TransactionsP, suite.Transaction)
+	}
+}
+
+func (suite *TransactionHistoryTestSuite) TestGetTransactionById() {
+	store := store.NewTransactionHistoryStoreModel(suite.Db)
+	var err error
+	var transaction *[]entities.TransactionHistory
+	var transact []*entities.TransactionHistory
+	var tr *entities.TransactionHistory
+	for _, value := range suite.TransactionsP {
+		transaction, err = store.GetTransactionsById(value.UserId)
+		if err != nil {
+			suite.T().Fatal("Unable to run InsertTransactionHistory store func")
+		}
+		for i := range *transaction {
+			tr = &(*transaction)[i]
+		}
+		transact = append(transact, tr)
+	}
+
+	for i, value := range transact {
+		suite.Equal(value.Id, transact[i].Id)
```

Comment:
@vpatcev18 this is not a valid test since you check the same object from the slice. You need to compare against `suite.Transactions` since those are the expected result.

Reply:
@FilipNikolovski resolved, plus solved the problem of multiple transactions on a single user